DuplicateFinderEngine/ScanEngine.cs

```diff
@@ -117,151 +126,154 @@ public class ScanEngine {
-					while (pauseTokenSource.IsPaused) {
-						Thread.Sleep(50);
-					}
+				var parallelOpts = new ParallelOptions {
+					MaxDegreeOfParallelism = Environment.ProcessorCount,
+					CancellationToken = cancelToken
+				};
+
+				var reScanList = ScanFileList
+					.Where(vf => !vf.Flags.Any(EntryFlags.ManuallyExcluded | EntryFlags.AllErrors))
+					.Where(vf => (vf.mediaInfo == null && !vf.IsImage) || vf.grayBytes == null)
+					.ToList();
+
+				InitProgress(reScanList.Count);
+				Parallel.For(0, reScanList.Count, parallelOpts, i => {
+					while (pauseTokenSource.IsPaused) Thread.Sleep(50);
 					var entry = reScanList[i];
 
 					if (entry.mediaInfo == null && !entry.IsImage) {
 						var ffProbe = new FFProbeWrapper.FFProbeWrapper();
 						var info = ffProbe.GetMediaInfo(entry.Path);
-						if (info == null) return;
-						entry.mediaInfo = info;
-
+						if (info == null)
+							entry.Flags.Set(EntryFlags.MetadataError);
+						else
+							entry.mediaInfo = info;
 					}
 
 					if (entry.grayBytes == null) {
-						entry.grayBytes = entry.IsImage ? GetImageAsBitmaps(entry, positionList.Count) : GetVideoThumbnailAsBitmaps(entry, positionList);
-						if (entry.grayBytes == null) return;
+						var (error, grayBytes) = entry.IsImage ? GetImageAsBitmaps(entry, positionList.Count) : GetVideoThumbnailAsBitmaps(entry, positionList);
+						if (error > 0)
+							entry.Flags.Set(error);
+						else
+							entry.grayBytes = grayBytes;
 					}
 
-					//report progress
-					IncrementProgress(remaining =>
-						Progress?.Invoke(this,
-							new OwnScanProgress {
-								CurrentPosition = processedFiles,
-								CurrentFile = entry.Path,
-								Elapsed = ElapsedTimer.Elapsed,
-								Remaining = remaining
-							}));
+					IncrementProgress(entry.Path);
 				});
 				st.Stop();
 				Logger.Instance.Info(string.Format(Properties.Resources.ThumbnailsFinished, st.Elapsed, processedFiles));
-				processedFiles = 0;
-				ScanProgressMaxValue = ScanFileList.Count;
-				st.Restart();
-				startTime = DateTime.Now;
 
+				st.Restart();
 				var percentageDifference = 1.0f - Settings.Percent / 100f;
-				
-				Parallel.For(0, ScanFileList.Count,
-					new ParallelOptions {
-						MaxDegreeOfParallelism = Environment.ProcessorCount,
-						CancellationToken = cancelToken
-					},
-					i => {
-						while (pauseTokenSource.IsPaused) {
-							Thread.Sleep(50);
+				var dupeScanList = ScanFileList.Where(vf => !vf.Flags.Any(EntryFlags.AllErrors | EntryFlags.ManuallyExcluded)).ToList();
+
+				InitProgress(dupeScanList.Count);
+				Parallel.For(0, dupeScanList.Count, parallelOpts, i => {
+					while (pauseTokenSource.IsPaused) Thread.Sleep(50);
+
+					var baseItem = dupeScanList[i];
+					if (baseItem.grayBytes == null || baseItem.grayBytes.Count == 0) {
+						IncrementProgress(baseItem.Path);
+						return;
+					}
+
+					for (var n = i + 1; n < dupeScanList.Count; n++) {
```

Comment:
Using i + 1 if the current processing file is near the end, it will only be scanned against a handful of items. You would miss lots of potential dups as the file is only checked against a handful of items then. Did I miss something?